neuraxle/pipeline.py

```diff
@@ -592,4 +592,35 @@ def join_fit_transform(self, step: Pipeline, data_container: DataContainer, cont
                 data_container_batch
             )
 
-        return step, output_data_container
\ No newline at end of file
+        return step, output_data_container
+
+
+class FullExecutionCheckpoint(ResumablePipeline):
+    """
+    Full checkpoint pipeline that handles every execution mode.
+
+    #. Saves data checkpoints for data inputs.
+    #. Saves data checkpoints for expected outputs.
+    #. Saves fitted step
+
+    .. seealso::
+        :class:`neuraxle.pipeline.ResumablePipeline`,
+        :class:`neuraxle.flow.TransformOnlyWrapper`,
+        :class:`neuraxle.flow.FitOnlyWrapper`,
+        :class:`neuraxle.flow.FitTransformOnlyWrapper`,
+        :class:`neuraxle.flow.StepSavingCheckpointer`
+    """
+
+    def __init__(self):
+        Pipeline.__init__(self, [
+            ('TransformOnlyWrapperDi', TransformOnlyWrapper(DataInputCheckpointerWrapper(PickleDataCheckpointer()))),
+            ('TransformOnlyWrapperEo', TransformOnlyWrapper(ExpectedOutputCheckpointerWrapper(PickleDataCheckpointer()))),
+
+            ('FitOnlyWrapperDi', FitOnlyWrapper(DataInputCheckpointerWrapper(PickleDataCheckpointer()))),
+            ('FitOnlyWrapperEo', FitOnlyWrapper(ExpectedOutputCheckpointerWrapper(PickleDataCheckpointer()))),
+
+            ('FitTransformOnlyWrapperDi', FitTransformOnlyWrapper(DataInputCheckpointerWrapper(PickleDataCheckpointer()))),
+            ('FitTransformOnlyWrapperEo', FitTransformOnlyWrapper(ExpectedOutputCheckpointerWrapper(PickleDataCheckpointer()))),
```

Comment:
Not sure if the fit_transform wrappers are needed, is rlly needed? It's getting complicated. Might revisit that later. Maybe fit_transform could be covered by the other wrappers